src/OptionSettlement.sol

```diff
@@ -21,27 +22,34 @@ import "./TokenURIGenerator.sol";
 // TODO(DRY code during testing)
 // TODO(Gas Optimize)
 
-// @notice This settlement protocol does not support rebase tokens, or fee on transfer tokens
-
+/// @notice This settlement protocol does not support rebase tokens, or fee on transfer tokens
 contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
-    // The protocol fee
+    /// @notice The protocol fee
     uint8 public immutable feeBps = 5;
 
-    // The address fees accrue to
+    /// @notice The address fees accrue to
     address public feeTo = 0x36273803306a3C22bc848f8Db761e974697ece0d;
 
-    // Fee balance for a given token
+    /// @notice Fee balance for a given token
     mapping(address => uint256) public feeBalance;
 
-    // The list of claims for an option
-    mapping(uint160 => uint256[]) internal unexercisedClaimsByOption;
-
-    // Accessor for Option contract details
+    /// @notice Accessor for Option contract details
     mapping(uint160 => Option) internal _option;
 
-    // Accessor for claim ticket details
+    /// @notice Accessor for claim ticket details
     mapping(uint256 => Claim) internal _claim;
 
+    /// @notice Accessor for buckets of claims grouped by day
+    /// @dev This is to enable O(constant) time options exercise. When options are written,
+    /// the Claim struct in this mapping is updated to reflect the cumulative amount written
+    /// on the day in question (block.timestamp/# seconds in a day). This allows
+    /// exercise() to assign options on each bucket from 'today' up until option expiry,
+    /// rather than a costly iteration of every unexercised claim on an option.
+    mapping(uint160 => mapping(uint16 => ClaimBucket)) internal _claimBucketByOptionAndDay;
+
+    /// @notice Accessor for mapping a claim id to its bucketed day
+    mapping(uint256 => ClaimIndex[]) internal _claimIdToClaimIndexArray;
```

Comment:
Need to update this natspec. Should we have an explicit access / view for this claimIndex, do you think @0xAlcibiades ?

Reply:
What would the use case be?

Reply:
I can't think of one. was curious if you could. can leave it out if there's not one.